### PR TITLE
Add test for maximum number of read_write image arguments in a kernel

### PR DESCRIPTION
**Pull request by shajder**
Related to #594 and [this](https://github.com/KhronosGroup/OpenCL-CTS/pull/2632#discussion_r3882168173) discussion

Adds test_api min_max_read_write_image_args, covering CL_DEVICE_MAX_READ_WRITE_IMAGE_ARGS the same way the existing tests cover the read_only and write_only limits.

[run-test: test_api min_max_read_write_image_args]

### CODE REVIEW

File: test_conformance/api/test_api_min_max.cpp [hunk 1]
```diff
@@ -558,6 +566,162 @@ REGISTER_TEST(min_max_write_image_args)
...
+    imageFormatDesc.image_channel_data_type = CL_FLOAT;
+
+    if (maxReadWriteImageArgs == 0)
+    {
+        log_info("WARNING: Device reports 0 for a max read/write image arg "
+                 "count (read/write image arguments unsupported). Skipping "
+                 "test (implicitly passes). This is only valid if the number "
+                 "of image formats is also 0.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
```
Comment (rjodinchr) [inline test_conformance/api/test_api_min_max.cpp]:
```suggestion
    if (get_device_cl_version(device) >= Version(3, 0) && maxReadWriteImageArgs == 0)
    {
        log_info("Device does not support read-write images. Skipping test.\n");
        return TEST_SKIPPED_ITSELF;
    }
```
Read-write images are mandatory in OpenCL 2.x when images are supported, but optional in OpenCL 3.0. Skipping the test when `maxReadWriteImageArgs == 0` should therefore be gated by `get_device_cl_version(device) >= Version(3, 0)`, so that non-conformant 2.x devices fail rather than skip.

File: test_conformance/api/test_api_min_max.cpp [hunk 2]
```diff
@@ -28,6 +28,14 @@ const char *sample_single_param_kernel[] = {
     "}\n"
 };
 
+const char *sample_read_write_image_kernel_pattern[] = {
```
Comment (rjodinchr) [inline test_conformance/api/test_api_min_max.cpp]:
```suggestion
static const char *sample_read_write_image_kernel_pattern[] = {
```

File: test_conformance/api/test_api_min_max.cpp [hunk 3]
```diff
@@ -558,6 +566,162 @@ REGISTER_TEST(min_max_write_image_args)
...
+                 "count (read/write image arguments unsupported). Skipping "
+                 "test (implicitly passes). This is only valid if the number "
+                 "of image formats is also 0.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    if (maxReadWriteImageArgs < minRequiredReadWriteImages)
+    {
+        log_error("ERROR: Reported max read/write image arg count is less than "
+                  "required! (%d)\n",
```
Comment (rjodinchr) [inline test_conformance/api/test_api_min_max.cpp]:
```suggestion
                  "required! (%u)\n",
```

**Review by rjodinchr — COMMENTED** [review]
Could you trigger a pocl run for it please?

Also I understand that this is mainly a copy/paste from another test, but keeping the declaration of `i` at function scope instead of inside the loop, and other similar things could be modernize as this is a new test.

File: test_conformance/api/test_api_min_max.cpp (hunk 1 above)
Comment (shajder) [reply test_conformance/api/test_api_min_max.cpp]: good find, corrected

File: test_conformance/api/test_api_min_max.cpp (hunk 2 above)
Comment (shajder) [reply test_conformance/api/test_api_min_max.cpp]: Done, though it looks odd next to the other non-static literals. Should we put all of them in an anonymous namespace?

File: test_conformance/api/test_api_min_max.cpp (hunk 3 above)
Comment (shajder) [reply test_conformance/api/test_api_min_max.cpp]: Done

**Comment by shajder** [comment]
> Also I understand that this is mainly a copy/paste from another test, but keeping the declaration of `i` at function scope instead of inside the loop, and other similar things could be modernize as this is a new test.

Fixed the loop counters. I kept the rest matching the surrounding code, since a single test written differently from the other 2000 lines in the file is harder to follow than a consistent one. Happy to modernize the whole file if that is something we want, but that feels like a separate PR.

File: test_conformance/api/test_api_min_max.cpp (hunk 2 above)
Comment (rjodinchr) [reply test_conformance/api/test_api_min_max.cpp]: I would say no.
If we want to clean up that code, let's have a PR dedicated for that.

**Review by rjodinchr — DISMISSED** [review]

**Comment by bashbaug** [comment]
Adding "focused review". Note, also has a merge conflict that needs to be resolved.

**Comment by shajder** [comment]
close/open to retrigger CI

**Review by rjodinchr — APPROVED** [review]
